scribe/lib/scribe/ai/presentation/widgets/ai_scribe.dart

```diff
@@ -0,0 +1,197 @@
+        final screenSize = MediaQuery.of(context).size;
+        const dialogWidth = barWidth;
+
+        // Calculate position 10px above the button
+        double left = buttonPosition.dx;
+        double bottom = screenSize.height - buttonPosition.dy + 10;
+
+        // Ensure dialog doesn't go off screen horizontally
+        if (left + dialogWidth > screenSize.width) {
+          left = screenSize.width - dialogWidth - 16;
+        }
+        if (left < 16) {
+          left = 16;
+        }
+
+        return Stack(
+          children: [
+            Positioned(
+              left: left,
+              bottom: bottom,
+              child: dialogContent,
+            ),
+          ],
+        );
+      }
+
+      return Center(child: dialogContent);
+    },
+  );
+
+  if (selectedAction == null) {
+    log('showAIScribeDialog: No action selected');
+    return;
+  }
+
+  if (!context.mounted) return;
+
+  await showDialog(
+    context: context,
+    barrierDismissible: true,
+    builder: (context) {
+      final suggestionFuture = _executeAIRequest(selectedAction, content);
+      final title = selectedAction.label;
+
+      final modalContent = AIScribeSuggestion(
+        title: title,
+        suggestionFuture: suggestionFuture,
+        onClose: () => Navigator.of(context).pop(),
+        onInsert: (result) {
+          onInsertText(result);
+          Navigator.of(context).pop();
+        },
+        imagePaths: imagePaths,
+      );
+
+      if (buttonPosition != null) {
+        // Get screen size to ensure modal stays within bounds
+        final screenSize = MediaQuery.of(context).size;
+        // Use responsive width: 90% on mobile, max 500px on larger screens
+        final modalWidth = screenSize.width < 600 ? screenSize.width * 0.9 : 500.0;
+        const modalHeight = 400.0;
+
+        // Calculate position 10px above the button
+        double left = buttonPosition.dx;
+        double bottom = screenSize.height - buttonPosition.dy + 10;
+
+        // Ensure modal doesn't go off screen horizontally
+        if (left + modalWidth > screenSize.width) {
+          left = screenSize.width - modalWidth - 16;
+        }
+        if (left < 16) {
+          left = 16;
+        }
+
+        // Ensure modal doesn't go off screen vertically
+        if (bottom + modalHeight > screenSize.height) {
+          bottom = screenSize.height - modalHeight - 16;
+        }
+        if (bottom < 16) {
+          bottom = 16;
+        }
+
+        return Stack(
+          children: [
+            Positioned(
+              left: left,
+              bottom: bottom,
+              child: modalContent,
+            ),
+          ],
+        );
+      }
+
+      return Center(child: modalContent);
+    },
+  );
+}
+
+Future<String> _executeAIRequest(
+  AIAction action,
+  String content,
+) async {
+  final dio = Dio();
+  final dataSource = AIDataSourceImpl(dio: dio);
+  final repository = AIScribeRepositoryImpl(dataSource);
```

Comment:
The presentation layer should not call classes from the data layer, use Dependency Injection (GetX)